src/core.jl

```diff
@@ -104,4 +104,25 @@ function match_keypoints(keypoints_1::Keypoints, keypoints_2::Keypoints, desc_1,
         end
     end
     matches
-end
\ No newline at end of file
+end
+
+"""
+```
+grade = grade_matches(keypoints_1, keypoints_2, difference_method)
+grade = grade_matches(keypoints_1, keypoints_2, limit, difference_method)
+```
+
+Returns the average difference between keypoints in `keypoints_1` and `keypoints_2` where
+`difference_method` is the method used for computing difference between individual pair of keypoints.
+If `limit` is provided then it returns the fraction of keypoint pairs which have
+`difference_method(keypoint_1,keypoint_2)` less than `limit`.
+"""
+
+function grade_matches(keypoints_1::Keypoints, keypoints_2::Keypoints, diff::Function = (i,j) -> (sqrt( (i[1]-j[1])^2 + (i[2]-j[2])^2 )))
+    @assert length(keypoints_1)==length(keypoints_2) "Keypoint lists are of different lengths."
+    @assert length(keypoints_1)!=0 "Keypoint list is of size zero."
+    mean(map((keypoint_1,keypoint_2)->diff(keypoint_1,keypoint_2), keypoints_1, keypoints_2))
+end
+
+grade_matches(keypoints_1::Keypoints, keypoints_2::Keypoints, limit::Real, diff::Function = (i,j)->(sqrt( (i[1]-j[1])^2 + (i[2]-j[2])^2 ))) =
```

Comment:
Do we even need this function? If we want it, it should return a `Bool` not an `Int`.